src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -0,0 +1,79 @@
+        {
+        }
+
+        internal override ImmutableHashSet<char> TriggerCharacters => CompletionUtilities.CommonTriggerCharactersWithArgumentList;
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem completionItem, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var declaration = root?.FindToken(completionListSpan.Start).GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (root is null || declaration is null)
+            {
+                return await base.GetChangeAsync(document, completionItem, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+            }
+
+            // MethodDeclarationSyntax, LocalFunctionStatementSyntax, AnonymousMethodExpressionSyntax, ParenthesizedLambdaExpressionSyntax, SimpleLambdaExpressionSyntax.
+
+            var newDeclaration = AddAsyncModifier(declaration, SyntaxGenerator.GetGenerator(document));
+            var newDocument = document.WithSyntaxRoot(root.ReplaceNode(declaration, newDeclaration));
+            var changes = await newDocument.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false);
+            var newText = await newDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            return CompletionChange.Create(CodeAnalysis.Completion.Utilities.Collapse(newText, changes.ToImmutableArray()));
+        }
+
+        private static SyntaxNode AddAsyncModifier(SyntaxNode declaration, SyntaxGenerator generator)
+        {
+            var modifiers = generator.GetModifiers(declaration);
+            return generator.WithModifiers(declaration, modifiers.WithAsync(true));
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var document = context.Document;
+            var position = context.Position;
+            var cancellationToken = context.CancellationToken;
+
+            var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
+
+            var workspace = document.Project.Solution.Workspace;
+            var syntaxContext = CSharpSyntaxContext.CreateContext(workspace, semanticModel, position, cancellationToken);
+
+            if (!syntaxContext.IsAwaitKeywordContext(position))
+            {
+                return;
+            }
+
+            var method = syntaxContext.TargetToken.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (method is not null && !method.GetModifiers().Any(SyntaxKind.AsyncKeyword))
+            {
+                context.AddItem(CommonCompletionItem.Create("await", string.Empty, CompletionItemRules.Default, inlineDescription: "Make container async"));
```

Comment:
```csharp
namespace MyNamespace
{
    public class Class1
    {    
        public void M()
        {
            awai$$
        }
    }

}
```

The completion list in the previous snippet shows this new completion item, but when I hit a tab or space, ONLY the async keyword is added to the method, but await doesn't get completed.

Not sure what could the problem be. It should be something simple.